File: story/lib/Audio.lua
---meta

---@class Audio
Audio={}
---注册BGM文件
---@param urls {id:string,url:string}[] 文件链接数组
function Audio.loadBGMFiles(urls)end
---注册SFX文件
---@param urls {id:string,url:string}[] 文件链接数组
function Audio.loadSFXFiles(urls)end
---卸载指定BGM
---@param id string 唯一标识符
function Audio.unloadBGM(id)end
---卸载指定SFX
---@param id string 唯一标识符
function Audio.unloadSFX(id)end
---预载BGM
---@param id string 唯一标识符
function Audio.preloadBGM(id)end
---预载SFX
---@param id string 唯一标识符
function Audio.preloadSFX(id)end
---卸载BGM
---@param id string 唯一标识符
function Audio.releaseBGM(id)end
---卸载SFX
---@param id string 唯一标识符
function Audio.releaseSFX(id)end
---将音频压入BGM混合层
---@param id string 唯一标识符
function Audio.mixBGM(id)end
---将音频压入SFX混合层
---@param id string 唯一标识符
function Audio.mixSFX(id)end
---播放BGM
---@param id string 唯一标识符
---@param data {volume:number,
---loop:boolean,fadeIn:number ,startTime:number} 播放参数
function Audio.playBGM(id,data)end
---播放SFX
---@param id string 唯一标识符
---@param data {volume:number,
---loop:boolean,fadeIn:number ,startTime:number} 播放参数
function Audio.playSFX(id,data)end
---把BGM移出混合层
---@param id string 唯一标识符
function Audio.removeBGM(id)end
---把SFX移出混合层
---@param id string 唯一标识符
function Audio.removeSFX(id)end

return Audio

File: src/server/ipc/audio.ts
import { BrowserWindow } from "electron";
import { win } from "../../main.js";

export class AudioIPCSender{
    // private Mywin:BrowserWindow
    constructor(window) {
        // this.win=window
    }
    public static loadBGMFiles(bgms){
        win.webContents.send('load-bgm-files',bgms)
    }
    public static loadSFXFiles(id){
        win.webContents.send('load-sfx-files',id)
    }
    public static unloadBGM(id){
        win.webContents.send('unload-bgm',id)
    }
    public static unloadSFX(id){
        win.webContents.send('unload-sfx',id)
    }
    public static preloadBGM(id){
        win.webContents.send('preload-bgm',id)
    }
    public static preloadSFX(id){
        win.webContents.send('preload-sfx',id)
    }
    public static releaseBGM(id){
        win.webContents.send('release-bgm',id)
    }
    public static releaseSFX(id){
        win.webContents.send('release-sfx',id)
    }
    public static mixBGM(id){
        win.webContents.send('mix-bgm',id)
    }
    public static mixSFX(id){
        win.webContents.send('mix-sfx',id)
    }
    public static playBGM(id,data){
        win.webContents.send('play-bgm',id,data)
    }
    public static playSFX(id,data){
        win.webContents.send('play-sfx',id,data)
    }
    public static removeBGM(id){
        win.webContents.send('remove-bgm',id)
    }
    public static removeSFX(id){
        win.webContents.send('remove-sfx',id)
    }
}